openstacksdk_plugin/resources/network/router.py

```diff
@@ -42,5 +42,35 @@ def remove_interface_from_router(openstack_resource, **kwargs):
     openstack_resource.remove_interface(kwargs)
 
 
-def update():
-    pass
+@with_openstack_resource(OpenstackRouter)
+def start(openstack_resource, **kwargs):
+    if kwargs and kwargs.get('routes'):
+        # Store routes in order to use them later on in order to remove them
+        # when the stop operation for router trigger
+        ctx.instance.runtime_properties['routes'] = kwargs['routes']
+        routes = dict()
+        routes['routes'] = kwargs['routes']
+        openstack_resource.update(routes)
+
+
+@with_openstack_resource(OpenstackRouter)
+def stop(openstack_resource, **kwargs):
+    if 'routes' in ctx.instance.runtime_properties:
+        # There are some routes need to be deleted since it is part of the
+        # runtime properties
+
+        # Routes need to be removed
+        routes_to_delete = ctx.instance.runtime_properties['routes']
```

Comment:
Can we find a more specific way to store routes? Maybe a set of route changes? The start and stop operation and store a 'default' set, but we should also be able to add others and identify them some how.

Reply:
So basically what are we doing her is to get the saved routes and then compare the routes required to be deleted in order to remove them all and keep the already existing one without deleting them, we store routes as runtime porp her, can you explain more about this ?

Reply:
Let's say someone uses this node. They add route X. Then during update they add route Y. Will stop remove only route X or route and Route Y?

Reply:
right now we are not supporting update operation for router and we are just depend on create/start/stop/delete in case we are going to add update operation then I will take into consideration if there are routes are being added to the node during update process, so then we need to update the  runtime_properties for `routes` so that in stop operation to remove all added/updated routes thats what I'm thinking of